android/app/src/main/java/app/organicmaps/settings/SettingsPrefsFragment.java

```diff
@@ -257,11 +257,10 @@ private void initSearchPrivacyPrefsCallbacks()
     final boolean isHistoryEnabled = Config.isSearchHistoryEnabled();
     ((TwoStatePreference) pref).setChecked(isHistoryEnabled);
     pref.setOnPreferenceChangeListener((preference, newValue) -> {
```

Comment:
```suggestion
    pref.setOnPreferenceChangeListener((preference, Boolean newValue) -> {
```

Will it compile? Then you don't need a (Boolean) cast below.

Reply:
No we cannot do that because its required data type is of type "Object".

Reply:
I see, poor Java... Then your solution with an intermediate casted variable is cleaner, you were right )

Reply:
i reverted the last commit. Please have a look once again : )